### ISSUE TITLE
[MangaSee] Unexpected JSON Token

### ISSUE THREAD

**Issue by CedMIX-Productions** [tachiyomiorg/tachiyomi-extensions#14253]
### Source information and language

MangaSee 1.3.34

### Steps to reproduce

1. Select a MangaSee source manga
2. Tap on Download on the latest chapter (downloading any chapter will do the same)
3. Go to More -> Downloads (Checked on More to see if it is stopped and it is)
![image](https://user-images.githubusercontent.com/31158494/201571194-51158c0d-3526-47fe-8610-5988271881be.png)
4. The MangaSee downloaded shows no progress at all
![image](https://user-images.githubusercontent.com/31158494/201572790-bf30a0d1-804b-4e91-bf77-52793320f792.png)

### Expected behavior

The download should finish without any errors in the notification tray and the status should not be paused.

### Actual behavior

When downloading any chapter, I immediately received this JSON error on the notification tray and the mangasee download is paused.

- Notifications
![image](https://user-images.githubusercontent.com/31158494/201573060-2fe085d0-8a56-4ada-9b38-2d32d2f1610d.png)

- Downloads
![image](https://user-images.githubusercontent.com/31158494/201573101-03466f57-0ad9-474b-ac15-2dca69aa365c.png)

### Tachiyomi version

0.14.2

### Android version

Android 12

### Other details

I did some digging by enabling "Verbose Logging" to print Tachiyomi's log into logcat. Since I use WSA (and Windows 11), I started a terminal (with powershell) and typed `adb logcat > .\tachiyomiLog.log` (after connecting to it by using `adb connect localhost:58526`. Once the log file is created, I used a terminal (this time with cmd) in order to run this command `type tachiyomiLog.log | findstr "11-14 12"` (11-14 is the date today, 12 is the current hour). I found the part that I need to show full error log.

The full error log from that find string filter (just essential to Tachiyomi only):

> 11-14 12:05:07.230 10937 11738 I okhttp.OkHttpClient: --> GET https://mangasee123.com/read-online/MARRIAGETOXIN/-chapter-1.html h2
11-14 12:05:07.230 10937 11738 I okhttp.OkHttpClient: Referer: https://mangasee123.com/
11-14 12:05:07.231 10937 11738 I okhttp.OkHttpClient: User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:71.0) Gecko/20100101 Firefox/77.0
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: Cache-Control: max-age=600
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: Host: mangasee123.com
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: Connection: Keep-Alive
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: Accept-Encoding: gzip
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: Cookie: PHPSESSID=m00id3c7ggi3hrl5fu4dt8t6j1
11-14 12:05:07.232 10937 11738 I okhttp.OkHttpClient: --> END GET
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: <-- 200 https://mangasee123.com/read-online/MARRIAGETOXIN/-chapter-1.html (354ms)
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: date: Mon, 14 Nov 2022 04:05:09 GMT
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: content-type: text/html; charset=UTF-8
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: expires: Thu, 19 Nov 1981 08:52:00 GMT
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: cache-control: no-store, no-cache, must-revalidate
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: pragma: no-cache
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: vary: Accept-Encoding
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: cf-cache-status: DYNAMIC
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=DWCAOPdvCPKiwmzh9b8PLgvEEXOpWRajxEiSmGzgFCXKvkQOUPyBhWqWf4EmswjtY3m5SYOXlfvOsCoPNCVeDM5vs4PLf9Vb9ZY2vgmcRJVdz1k73mj6u5S2%2BKfyPHwWZ3c%3D"}],"group":"cf-nel","max_age":604800}
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: server: cloudflare
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: cf-ray: 769cddfa0d1724bf-HKG
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: content-encoding: gzip
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
11-14 12:05:07.586 10937 11738 I okhttp.OkHttpClient: <-- END HTTP
11-14 12:05:07.636 10937 11738 E Downloader: kotlinx.serialization.json.internal.JsonDecodingException: Unexpected JSON token at offset 14: Expected EOF after parsing, but had M instead at path: $
11-14 12:05:07.636 10937 11738 E Downloader: JSON input:
11-14 12:05:07.636 10937 11738 E Downloader:            function MainFunction($http){
11-14 12:05:07.636 10937 11738 E Downloader:                    var vm = this
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.internal.AbstractJsonLexer.fail(AbstractJsonLexer.kt:68)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.internal.AbstractJsonLexer.fail$default(AbstractJsonLexer.kt:15)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.internal.AbstractJsonLexer.expectEof(AbstractJsonLexer.kt:45)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.Json.decodeFromString(Json.kt:35)
11-14 12:05:07.636 10937 11738 E Downloader:    at kotlinx.serialization.json.Json.parseToJsonElement(Json.kt:8)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.multisrc.nepnep.NepNep.pageListParse(NepNep.kt:325)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.source.online.HttpSource$fetchPageList$1.invoke(HttpSource.kt:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.source.online.HttpSource$fetchPageList$1.invoke(HttpSource.kt:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.source.online.HttpSource$$ExternalSyntheticLambda0.call(R8$$SyntheticClass:8)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:3)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach$DoOnEachSubscriber.onNext(OnSubscribeDoOnEach.java:13)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.network.OkHttpExtensionsKt$asObservable$1$requestArbiter$1.request(OkHttpExtensions.kt:32)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Subscriber.setProducer(Subscriber.java:39)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:3)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Subscriber.setProducer(Subscriber.java:22)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:3)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:3)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Subscriber.setProducer(Subscriber.java:22)
11-14 12:05:07.636 10937 11738 E Downloader:    at eu.kanade.tachiyomi.network.OkHttpExtensionsKt$$ExternalSyntheticLambda1.call(R8$$SyntheticClass:24)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:4)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:2)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDoOnEach.call(OnSubscribeDoOnEach.java:1)
11-14 12:05:07.636 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:4)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:1)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:3)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:1)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.Observable.unsafeSubscribe(Observable.java:10)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:12)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:12)
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:10)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-14 12:05:07.637 10937 11738 E Downloader:    at java.lang.Thread.run(Thread.java:920)
11-14 12:05:07.637 10937 11738 E Downloader: Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: okhttp3.Response.class
11-14 12:05:07.637 10937 11738 E Downloader:    at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:20)
11-14 12:05:07.637 10937 11738 E Downloader:    ... 59 more
11-14 12:05:07.734 10937 10960 I anade.tachiyom: Background young concurrent copying GC freed 147454(6532KB) AllocSpace objects, 13(364KB) LOS objects, 41% free, 10011KB/16MB, paused 112us,35us total 107.991ms
11-14 12:05:07.845   551   551 D NotificationListener: New notification detected by a notification listener
11-14 12:05:07.845   551   551 D NotificationListener: Notification audioAttribute Usage USAGE_NOTIFICATION contentType CONTENT_TYPE_SONIFICATION
11-14 12:05:07.845   551   551 D NotificationRedirectionHandler: Generating new notification message..
11-14 12:05:07.845   551   551 D NotificationRedirectionHandler: Notification package name:eu.kanade.tachiyomi, id:-202, tag:null, app name:Tachiyomi, title:MARRIAGETOXIN: Chapter 1, text:Unexpected JSON token at offset 14: Expected EOF after parsing, but had M instead at path: $
11-14 12:05:07.845   551   551 D NotificationRedirectionHandler: JSON input:
11-14 12:05:07.845   551   551 D NotificationRedirectionHandler:                function MainFunction($http){
11-14 12:05:07.845   551   551 D NotificationRedirectionHandler:                        var vm = this
11-14 12:05:07.849   551   551 D NotificationRedirectionHandler: sendNotification sent notification message.

Device used: WSA version 2209.40000.26.0 with Android 12

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.2](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

**Comment by curche** [comment]
Checking the same chapter
The new link appears to be
https://mangasee123.com/read-online/MARRIAGETOXIN-chapter-1.html

Which already worked in the extension at my end

Try the following:
1. Remove manga from library
2. Go to More > Settings > Advanced and tap Clear Cache and Clear Database
3. Go back to Browse > Source > MangaSee and find and try reading the chapter again

**Comment by CedMIX-Productions** [comment]
The tips you suggested works now (11-15-2022 8:45 AM PST). I even reverted my user-agent to default (the last user agent I used was from Firefox for Windows). I might go ahead to restore my library really quick and close this issue as resolved later once testing is complete.
Update note: Testing is now complete. Closing this issue as **resolved**.
